concordium-sdk/src/main/java/com/concordium/sdk/Client.java

```diff
@@ -313,6 +312,25 @@ public Semver getVersion() {
         return new Semver(versionString);
     }
 
+    /**
+     * Get the information about total amount of CCD and the state of various administrative accounts.
+     * @param blockHash Block at which the reward status is to be retrieved.
+     * @return Parsed {@link RewardsOverview}.
+     * @throws Exception When the returned response is null.
+     */
+    public RewardsOverview getRewardStatus(final Hash blockHash) throws Exception {
```

Comment:
Please use `BlockNotFoundException`

Reply:
done

Reply:
Please also reflect it in the function signature. Currently it `throws Exception` but should be `throws BlockNotFoundException` .